src/main/java/mate/academy/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,52 @@
+package mate.academy.dao.impl;
+
+import java.util.Optional;
+import mate.academy.dao.UserDao;
+import mate.academy.model.User;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+public class UserDaoImpl implements UserDao {
+    @Override
+    public User save(User user) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
```

Comment:
Consider moving the `SessionFactory` initialization outside of the method to a class-level variable. This will prevent the unnecessary creation of a `SessionFactory` instance every time a method is called, which can be resource-intensive.